applications/CoSimulationApplication/python_scripts/convergence_accelerators/iqnils.py

```diff
@@ -143,7 +143,7 @@ def UpdateSolution( self, r, x ):
     ## FinalizeSolutionStep()
     # Finalizes the current time step and initializes the next time step.
     def FinalizeSolutionStep( self ):
-        if self.V_new != [] and self.W_new != []:
+        if len(self.V_new) != 0 and len(self.W_new) != 0:
```

Comment:
Just use if self.V_new, checking len() == 0 is an antipattern afaik

Reply:
I'd do it, but numpy is telling me not to.

```py
import numpy
a = numpy.array([])
bool(a)
```
```
<python-input-3>:1: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
  bool(a)
False
```

Reply:
ah I thought it was a `list`
All good then

Reply:
out of curiosity, the message suggests to use `array.size > 0`, why did you go with the `len` based check? Did you experience problems? (I also use numpy and would like to know)